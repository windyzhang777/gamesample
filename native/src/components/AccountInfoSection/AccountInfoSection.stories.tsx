import React from 'react';
import {storiesOf} from '@storybook/react-native';
import AccountInfoSection from '.';

const demoProfileImage = require('../../assets/images/demoProfileImage.png');
storiesOf('Account Info Section', module).add('Normal Account Info Section', () => {
  return <AccountInfoSection username={'G-MOM35'} profileImageSource={demoProfileImage} />;
});
