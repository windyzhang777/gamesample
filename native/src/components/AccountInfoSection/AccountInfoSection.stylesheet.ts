import {StyleSheet} from 'react-native';
import {navigationPrimaryColor} from 'shared/theme/variables';

export default StyleSheet.create({
  AccountInfoSection: {
    backgroundColor: navigationPrimaryColor,
    paddingTop: 20,
    paddingBottom: 10,
    flexDirection: 'row',
    alignItems: 'center',
  },
  AccountInfoSectionProfileImage: {
    flex: 1,
    width: undefined,
    height: undefined,
  },
  AccountInfoSectionProfileImageContainer: {
    width: 75,
    height: 75,
    marginRight: 20,
  },
  AccountInfoSectionUsername: {
    fontWeight: 'bold',
    fontSize: 18,
    color: 'white',
  },
});
