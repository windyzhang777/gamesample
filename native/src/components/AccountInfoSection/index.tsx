import React from 'react';
import AccountInfoSectionShared, {
  AccountInfoSectionProps,
} from 'shared/containers/AccountInfoSection';
import {Text, View, Image} from 'react-native';
import styles from './AccountInfoSection.stylesheet';

/**
 * This is a native version of rendering
 * @param {AccountInfoSectionProps} props
 */
export default function AccountInfoSection(props: AccountInfoSectionProps) {
  //This is just passing down the properties
  return (
    <AccountInfoSectionShared
      {...props}
      renderComponent={({profileImageSource, username}: AccountInfoSectionProps) => {
        // This is where the native rendering is happening
        return (
          <View style={styles.AccountInfoSection}>
            <View style={styles.AccountInfoSectionProfileImageContainer}>
              <Image
                style={styles.AccountInfoSectionProfileImage}
                resizeMode={'contain'}
                source={profileImageSource}
              />
            </View>
            <Text style={styles.AccountInfoSectionUsername}>{username}</Text>
          </View>
        );
      }}
    />
  );
}
