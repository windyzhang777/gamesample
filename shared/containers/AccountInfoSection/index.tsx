import {RenderedComponent} from '../../models/RenderedComponent';

export interface AccountInfoSectionProps {
  profileImageSource: any;
  username: string;
}

export default function AccountInfoSection({
  profileImageSource,
  username,
  renderComponent,
}: AccountInfoSectionProps & RenderedComponent<AccountInfoSectionProps>) {
  const renderProps: AccountInfoSectionProps = {
    profileImageSource,
    username: username,
  };
  return renderComponent(renderProps);
}
