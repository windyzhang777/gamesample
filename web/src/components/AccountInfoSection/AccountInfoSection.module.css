.AccountInfoSection {
  display: flex;
  align-items: center;
}

.ProfileImage {
  font-size: inherit;
  margin-bottom: 20px;
}

.Username {
  font-size: inherit;
}
