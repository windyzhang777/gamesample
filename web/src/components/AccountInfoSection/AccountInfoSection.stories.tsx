import React from 'react';
import {storiesOf} from '@storybook/react';
import AccountInfoSection from '.';
import demoProfileImage from 'native/src/assets/images/demoProfileImage.png';

storiesOf('AccountInfoSection', module).add('Normal Account Info Section', () => {
  return <AccountInfoSection username={'G-MOM35'} profileImageSource={demoProfileImage} />;
});
