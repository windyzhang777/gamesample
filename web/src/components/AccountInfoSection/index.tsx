import React from 'react';
import AccountInfoSectionShared, {
  AccountInfoSectionProps,
} from 'shared/containers/AccountInfoSection';
import classes from './AccountInfoSection.module.css';

/**
 * This is a web version of rendering
 * @param {AccountInfoSectionProps} props
 */

export default function AccountInfoSection(props: AccountInfoSectionProps) {
  return (
    <AccountInfoSectionShared
      {...props}
      renderComponent={({profileImageSource, username}: AccountInfoSectionProps) => {
        return (
          <div className={classes.AccountInfoSection}>
            <div className={classes.ProfileImage}>
              <img
                style={{
                  objectFit: 'cover',
                  height: '70px',
                  marginRight: '10px',
                }}
                src={profileImageSource}
                alt="demoProfileImage"
              />
            </div>
            <div className={classes.Username}>{username}</div>
          </div>
        );
      }}
    />
  );
}
