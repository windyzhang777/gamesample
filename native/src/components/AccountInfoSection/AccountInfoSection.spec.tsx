import React from 'react';
import {shallow} from 'enzyme';
import AccountInfoSection from './index';

describe('<AccountInfoSection/>', () => {
  it('renders wrapper', () => {
    expect(shallow(<AccountInfoSection />)).toMatchSnapshot();
  });
  it('renders component', () => {
    const RenderPropComponent = shallow(<AccountInfoSection />).prop('renderComponent');
    expect(
      shallow(
        <RenderPropComponent
          profileImageSource={{uri: 'URL for profile image'}}
          username={'gmom-35'}
        />,
      ),
    ).toMatchSnapshot();
  });
});
