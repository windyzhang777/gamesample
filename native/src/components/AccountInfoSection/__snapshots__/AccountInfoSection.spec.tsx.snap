// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountInfoSection/> renders component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#2f3d86",
      "flexDirection": "row",
      "paddingBottom": 10,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "height": 75,
        "marginRight": 20,
        "width": 75,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "URL for profile image",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": undefined,
          "width": undefined,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
    gmom-35
  </Text>
</View>
`;

exports[`<AccountInfoSection/> renders wrapper 1`] = `
<AccountInfoSection
  renderComponent={[Function]}
/>
`;
